repository/src/main/java/org/apache/atlas/tasks/AtlasTaskService.java

```diff
@@ -124,6 +133,197 @@ public void retryTask(String taskGuid) throws AtlasBaseException {
         graph.commit();
     }
 
+    @Override
+    @GraphTransaction
+    public List<AtlasTask> createAtlasTasks(List<AtlasTask> tasks) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord("createAtlasTasks");
+        List<String> supportedTypes = new ArrayList<>();
+        // Get all the supported task type dynamically
+        Reflections reflections = new Reflections("org.apache.atlas.repository.store.graph.v2.tasks", new SubTypesScanner(false));
+        Set<Class<? extends TaskFactory>> classes = reflections.getSubTypesOf(TaskFactory.class);
+        for (Class<?> clazz : classes) {
+            List<Class<?>> interfaces = Arrays.asList(clazz.getInterfaces());
+            if (interfaces.contains(TaskFactory.class)) {
+                try {
+                    Field field = clazz.getField("supportedTypes");
+                    List<String> supportedTaskTypes = (List<String>) field.get(null);
+                    supportedTypes.addAll(supportedTaskTypes);
+                } catch (NoSuchFieldException | IllegalAccessException e) {
+                    LOG.error("Error occurred while accessing field: ", e);
+                    throw new AtlasBaseException("Error occurred while accessing field", e);
+                }
+            }
+        }
+        List<AtlasTask> createdTasks = new ArrayList<>();
+
+        if (CollectionUtils.isNotEmpty(tasks)) {
+            for (AtlasTask task : tasks) {
+                String taskType = task.getType();
+                if (!supportedTypes.contains(taskType)) {
+                    throw new AtlasBaseException(AtlasErrorCode.TASK_TYPE_NOT_SUPPORTED, task.getType());
+                }
+                if (isClassificationTaskType(taskType)) {
+                    String classificationName = task.getClassificationName();
+                    String entityGuid = task.getEntityGuid();
+                    String classificationId = resolveAndReturnClassificationId(classificationName, entityGuid);
+                    if (StringUtils.isEmpty(classificationId)) {
+                        throw new AtlasBaseException(AtlasErrorCode.TASK_INVALID_PARAMETERS, task.toString());
+                    }
+                    task.getParameters().put(PARAM_CLASSIFICATION_VERTEX_ID, classificationId);
+                }
+                task.setUpdatedTime(new Date());
+                task.setCreatedTime(new Date());
+                task.setStatusPending();
+                task.setAttemptCount(0);
+                task.setGuid(UUID.randomUUID().toString());
+                task.setCreatedBy(RequestContext.getCurrentUser());
+
+                createTaskVertex(task);
+                createdTasks.add(task);
+            }
+        }
+
+        graph.commit();
```

Comment:
Do we need graph.commit();, given we are using graphTransaction annotation

Reply:
Addressed